packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/services/PapayaObjectDuplicator.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.papaya.services;
+
+import java.util.Objects;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.papaya.NamedElement;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.web.application.views.explorer.services.DefaultObjectDuplicator;
+import org.eclipse.sirius.web.application.views.explorer.services.api.DuplicationSettings;
+import org.eclipse.sirius.web.application.views.explorer.services.api.IObjectDuplicator;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Custom object duplication logic for Papaya named elements to give a distinct name to the new copy.
+ *
+ * @author pcdavid
+ */
+@Service
+public class PapayaObjectDuplicator implements IObjectDuplicator {
```

Comment:
This is only used for a test, it can be moved in the test services. No need to impact the regular behavior every time.

Reply:
Done.